packages/runtime/src/widgets/SignInWithGoogle.tsx

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useMemo } from "react";
+import { GoogleLogin, GoogleOAuthProvider } from "@react-oauth/google";
+import type { CredentialResponse } from "@react-oauth/google";
+import { jwtDecode } from "jwt-decode";
+import { useRegisterBindings } from "@ensembleui/react-framework";
+import { useEnsembleUser } from "@ensembleui/react-framework";
+import type { EnsembleAction } from "@ensembleui/react-framework";
+import { WidgetRegistry } from "../registry";
+import { useEnsembleAction } from "../runtime/hooks/useEnsembleAction";
+
+export type SignInWithGoogleProps = {
+  type?: "standard" | "icon";
+  theme?: "outline" | "filled_blue" | "filled_black";
+  size?: "large" | "small" | "medium";
+  text?: "signin_with" | "signup_with";
+  shape?: "rectangular" | "pill" | "circle" | "square";
+  onSignedIn?: EnsembleAction;
+  onError?: EnsembleAction;
+};
+
+export const SignInWithGoogle: React.FC<SignInWithGoogleProps> = (props) => {
+  // TODO: need to move this inside env
+  const oAuthClientId =
+    "726646987043-9i1it0ll0neojkf7f9abkagbe66kqe4a.apps.googleusercontent.com";
```

Comment:
If you don't need this at the application context provider level, this can just be a prop in your widget.